### ISSUE TITLE
Apply Unify to all datasources.

### ISSUE THREAD

**Issue by fioddor** [bitergia-analytics/bitergia-analytics#96]
https://github.com/bitergia-analytics/bitergia-analytics/issues/59 limited the automatic unification of identities to "trusted" sources.

Please remove this limitation. The pair (user, source) is safe for all sources.

The limitation "_Also, these flag only applies to trusted datasources (i.e. github and gitlab)._ " was meant in the opposite way. 

Both github and gitlab have multiple datasources:

-  Github: github2:issue, github2:pull, github:issue, github:pull, github:repo
-  GitLab: gitlab:issue, gitlab:merge

The **disabled** switch should merge the same username from gerrit, git, github:issue, ...

The **enabled** switch applies the restriction and only allows to merge different datasources if they are trusted datasources:
- The same username in 2 identities both from `gerrit` should be merged according to the restriction because they both share the same datasource.
- The same user from `github:issue` and `github:repo` should be merged regardless of the restriction because they are both trusted to come from the same source (github).
- The same username from `gerrit` and `github:repo` should NOT be merged (applying the restriction) because `gerrit` is NOT a trusted datasource.
- The same username from `github:issue` and `gitlab:merge` should NOT be merged (applying the restriction) because they belong to different trusted datasources.

Blank/void/inexisting usernames should NOT be merged (regardless of the restriction status)

All this is meant for the usual case where there's a single origin instance per datasource. In case of multiple instances (like different hosted gitlab origin instances) the _Unify_ job shouldn't be used without an aware decission (for example: assuming a small error due to automatic unification over a greater error due to a huge number of non-unified identities).

**Comment by canasdiaz** [comment]
@fioddor Can you please provide an example of how this is affecting our efficiency when improving the identity data set? Remember to hide personal data.

**Comment by fioddor** [comment]
# Context

Bob is contributing to a tracked project with identities: `Bob bob@this.org` and `robert@company.com`.

The foundation provides a list of known emails for project members and both emails are in the list.
This results in 2 additional identities for Bob: `666 bob@this.org` and `666 robert@company.com`, being `666` Bob's id in the foundation.

Now _Unify_ unifies by email, so it merges `Bob bob@this.org` with `666 bob@this.org` in a first profile and `robert@company.com` with `666 robert@company.com` in a second profile.

# Problem/Limitation

Since Unify is restricted to merging usernames only for `gitlab` and `github` (or all the data sources as one), the profiles remain separated as if Bob and Robert were different individuals, despite both profiles share the userid `666` from the foundation's source.

**Comment by dicortazar** [comment]
I guess we needed to have the trusted data sources as if not, the platform start creating the big snowballs with identities from wherever. In other words, the removal of the flag for trusted data sources should be done under certain conditions (not always) and perhaps only by an administrator profile. Does this make sense?

**Comment by fioddor** [comment]
>  if not, the platform start creating the big snowballs with identities from wherever.

That's true for unifications ACROSS data sources. Here I'm only asking to unify users WITHIN each datasource.
